src/main/java/de/undercouch/actson/JsonParser.java

```diff
@@ -399,14 +402,40 @@ private void parse(char nextChar) {
         event1 = stateToEvent();
       }
 
-      // Change the state.
       state = nextState;
+    } else if (nextState == XU) {
```

Comment:
This would require to add nextChar to performAction parameters. 
Like `private void performAction(byte action, nextChar) { ... }`
Is this OK?